### ISSUE TITLE
Strange test failure on remote builder

### ISSUE THREAD

**Issue by sternenseemann** [vmchale/libarchive#20]
I haven't been able to figure out this test failure occurring on a builder of our [nixpkgs CI infrastructure](https://hydra.nixos.org/jobset/nixpkgs/haskell-updates). It happens consistently on different builders, but only on `x86_64-linux` ones. I can't reproduce it locally at all.

My best guess it has something to do with parallelization of the test in question maybe. The builder passes `-j16` everywhere, so I guess that is also what `hspec` detects. Not sure what of the test could not be threadsafe so far, had a look at `temporary` and it all seemed reasonable…

```
setupCompilerEnvironmentPhase
Build with /nix/store/jyny464p3c2g9dkq5y2fw2g4jgfvkm33-ghc-8.10.4.
unpacking sources
unpacking source archive /nix/store/h7p32kvi7zpgxvf8aqdgyphy0aqzhy42-libarchive-3.0.2.1.tar.gz
source root is libarchive-3.0.2.1
setting SOURCE_DATE_EPOCH to timestamp 1000000000 of file libarchive-3.0.2.1/test/Spec.cpphs
patching sources
compileBuildDriverPhase
setupCompileFlags: -package-db=/build/setup-package.conf.d -j16 +RTS -A64M -RTS -threaded -rtsopts
[1 of 1] Compiling Main             ( Setup.hs, /build/Main.o )
Linking Setup ...
configuring
configureFlags: --verbose --prefix=/nix/store/jw8fr6ggh8pqlayn7wnwm9hjr1kbkksh-libarchive-3.0.2.1 --libdir=$prefix/lib/$compiler --libsubdir=$abi/$libname --docdir=/nix/store/7c6vnfdfp6k2wml9s3ssyfwnajncqm79-libarchive-3.0.2.1-doc/share/doc/libarchive-3.0.2.1 --with-gcc=gcc --package-db=/build/package.conf.d --ghc-options=-j16 +RTS -A64M -RTS --disable-split-objs --enable-library-profiling --profiling-detail=exported-functions --disable-profiling --enable-shared --disable-coverage --enable-static --disable-executable-dynamic --enable-tests --disable-benchmarks --enable-library-vanilla --disable-library-for-ghci --ghc-option=-split-sections --extra-lib-dirs=/nix/store/qq9m6635nqabr19qwzh24dh755kamsnf-ncurses-6.2/lib --extra-lib-dirs=/nix/store/8dg6v2wzg7gsaah36wrj86mxvi3svq2b-libffi-3.3/lib --extra-lib-dirs=/nix/store/bqbsg99m3idjwwcchdyqg9ixslzx06mi-gmp-6.2.1/lib --extra-include-dirs=/nix/store/sh96v5wxf9dmwawm46ixqm0ksh0y3kqd-libarchive-3.5.1-dev/include --extra-lib-dirs=/nix/store/sh96v5wxf9dmwawm46ixqm0ksh0y3kqd-libarchive-3.5.1-dev/lib --extra-include-dirs=/nix/store/mmqipjmk2z2jmp25hhyngcxj6v5g88c5-attr-2.4.48-dev/include --extra-lib-dirs=/nix/store/mmqipjmk2z2jmp25hhyngcxj6v5g88c5-attr-2.4.48-dev/lib --extra-lib-dirs=/nix/store/k6pqxz3h0knzsicj8rs185ps5nz8gwsl-attr-2.4.48/lib --extra-include-dirs=/nix/store/jp1jxwkvr7xzxzk54jb4chh2n3viw7dx-acl-2.3.0-dev/include --extra-lib-dirs=/nix/store/jp1jxwkvr7xzxzk54jb4chh2n3viw7dx-acl-2.3.0-dev/lib --extra-lib-dirs=/nix/store/p75vg0nry7rl42fwf8yydnl7sj1j0grs-acl-2.3.0/lib --extra-lib-dirs=/nix/store/0ky3yw5sjprs09q79kw5yv1crfc1af82-libarchive-3.5.1-lib/lib
Using Parsec parser
Configuring libarchive-3.0.2.1...
Flags chosen: cross=False, low-memory=False, system-libarchive=True
Dependency base >=4.8 && <5: using base-4.14.1.0
Dependency bytestring -any: using bytestring-0.10.12.0
Dependency composition-prelude >=2.0.5.0: using composition-prelude-3.0.0.2
Dependency deepseq >=1.4.0.0: using deepseq-1.4.4.0
Dependency dlist -any: using dlist-0.8.0.8
Dependency filepath -any: using filepath-1.4.2.1
Dependency mtl >=2.2.1: using mtl-2.2.2
Dependency unix-compat >=0.1.2.1: using unix-compat-0.5.3
Dependency base -any: using base-4.14.1.0
Dependency bytestring -any: using bytestring-0.10.12.0
Dependency composition-prelude >=2.0.5.0: using composition-prelude-3.0.0.2
Dependency dir-traverse -any: using dir-traverse-0.2.3.0
Dependency directory >=1.2.5.0: using directory-1.3.6.0
Dependency filepath -any: using filepath-1.4.2.1
Dependency hspec -any: using hspec-2.7.10
Dependency libarchive -any: using libarchive-3.0.2.1
Dependency mtl >=2.2.2: using mtl-2.2.2
Dependency pathological-bytestrings -any: using
pathological-bytestrings-0.1.0.0
Dependency temporary -any: using temporary-1.3
Dependency libarchive (==3.5.0 || >3.5.0) && <4.0: using version 3.5.1
Source component graph:
    component lib
    component test:libarchive-test dependency lib
Configured component graph:
    component libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq
        include base-4.14.1.0
        include bytestring-0.10.12.0
        include composition-prelude-3.0.0.2-Iz0NadNXLWqCD8l3HZGWwK
        include deepseq-1.4.4.0
        include dlist-0.8.0.8-FJ4hncuTyhRHtpRjYcZTlQ
        include filepath-1.4.2.1
        include mtl-2.2.2
        include unix-compat-0.5.3-E98n0JyMac7Bz478oqWJHX
    component libarchive-3.0.2.1-885awQtzh6G10WUQE9hNAl-libarchive-test
        include base-4.14.1.0
        include bytestring-0.10.12.0
        include composition-prelude-3.0.0.2-Iz0NadNXLWqCD8l3HZGWwK
        include dir-traverse-0.2.3.0-Gl4x4Zfc2rV3pGFikKPEFP
        include directory-1.3.6.0
        include filepath-1.4.2.1
        include hspec-2.7.10-B5EHETeLi5oAFDXW4Lxz5e
        include libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq
        include mtl-2.2.2
        include pathological-bytestrings-0.1.0.0-CR4TnrQV2AzBlmMvk7ARhy
        include temporary-1.3-5rKowKsBN9cJXiXXFDM1ci
Linked component graph:
    unit libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq
        include base-4.14.1.0
        include bytestring-0.10.12.0
        include composition-prelude-3.0.0.2-Iz0NadNXLWqCD8l3HZGWwK
        include deepseq-1.4.4.0
        include dlist-0.8.0.8-FJ4hncuTyhRHtpRjYcZTlQ
        include filepath-1.4.2.1
        include mtl-2.2.2
        include unix-compat-0.5.3-E98n0JyMac7Bz478oqWJHX
        Codec.Archive=libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq:Codec.Archive,Codec.Archive.Foreign=libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq:Codec.Archive.Foreign,Codec.Archive.Foreign.Archive=libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq:Codec.Archive.Foreign.Archive,Codec.Archive.Foreign.ArchiveEntry=libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq:Codec.Archive.Foreign.ArchiveEntry,Codec.Archive.Internal.Monad=libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq:Codec.Archive.Internal.Monad,Codec.Archive.Internal.Pack=libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq:Codec.Archive.Internal.Pack,Codec.Archive.Internal.Pack.Common=libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq:Codec.Archive.Internal.Pack.Common,Codec.Archive.Internal.Pack.Lazy=libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq:Codec.Archive.Internal.Pack.Lazy,Codec.Archive.Internal.Unpack=libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq:Codec.Archive.Internal.Unpack,Codec.Archive.Internal.Unpack.Lazy=libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq:Codec.Archive.Internal.Unpack.Lazy
    unit libarchive-3.0.2.1-885awQtzh6G10WUQE9hNAl-libarchive-test
        include base-4.14.1.0
        include bytestring-0.10.12.0
        include composition-prelude-3.0.0.2-Iz0NadNXLWqCD8l3HZGWwK
        include dir-traverse-0.2.3.0-Gl4x4Zfc2rV3pGFikKPEFP
        include directory-1.3.6.0
        include filepath-1.4.2.1
        include hspec-2.7.10-B5EHETeLi5oAFDXW4Lxz5e
        include libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq
        include mtl-2.2.2
        include pathological-bytestrings-0.1.0.0-CR4TnrQV2AzBlmMvk7ARhy
        include temporary-1.3-5rKowKsBN9cJXiXXFDM1ci
Ready component graph:
    definite libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq
        depends base-4.14.1.0
        depends bytestring-0.10.12.0
        depends composition-prelude-3.0.0.2-Iz0NadNXLWqCD8l3HZGWwK
        depends deepseq-1.4.4.0
        depends dlist-0.8.0.8-FJ4hncuTyhRHtpRjYcZTlQ
        depends filepath-1.4.2.1
        depends mtl-2.2.2
        depends unix-compat-0.5.3-E98n0JyMac7Bz478oqWJHX
    definite libarchive-3.0.2.1-885awQtzh6G10WUQE9hNAl-libarchive-test
        depends base-4.14.1.0
        depends bytestring-0.10.12.0
        depends composition-prelude-3.0.0.2-Iz0NadNXLWqCD8l3HZGWwK
        depends dir-traverse-0.2.3.0-Gl4x4Zfc2rV3pGFikKPEFP
        depends directory-1.3.6.0
        depends filepath-1.4.2.1
        depends hspec-2.7.10-B5EHETeLi5oAFDXW4Lxz5e
        depends libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq
        depends mtl-2.2.2
        depends pathological-bytestrings-0.1.0.0-CR4TnrQV2AzBlmMvk7ARhy
        depends temporary-1.3-5rKowKsBN9cJXiXXFDM1ci
Using Cabal-3.2.1.0 compiled by ghc-8.10
Using compiler: ghc-8.10.4
Using install prefix:
/nix/store/jw8fr6ggh8pqlayn7wnwm9hjr1kbkksh-libarchive-3.0.2.1
Executables installed in:
/nix/store/jw8fr6ggh8pqlayn7wnwm9hjr1kbkksh-libarchive-3.0.2.1/bin
Libraries installed in:
/nix/store/jw8fr6ggh8pqlayn7wnwm9hjr1kbkksh-libarchive-3.0.2.1/lib/ghc-8.10.4/x86_64-linux-ghc-8.10.4/libarchive-3.0.2.1-6iG1yZb9GRSIaEGy26Fbmq
Dynamic Libraries installed in:
/nix/store/jw8fr6ggh8pqlayn7wnwm9hjr1kbkksh-libarchive-3.0.2.1/lib/ghc-8.10.4/x86_64-linux-ghc-8.10.4
Private executables installed in:
/nix/store/jw8fr6ggh8pqlayn7wnwm9hjr1kbkksh-libarchive-3.0.2.1/libexec/x86_64-linux-ghc-8.10.4/libarchive-3.0.2.1
Data files installed in:
/nix/store/jw8fr6ggh8pqlayn7wnwm9hjr1kbkksh-libarchive-3.0.2.1/share/x86_64-linux-ghc-8.10.4/libarchive-3.0.2.1
Documentation installed in:
/nix/store/7c6vnfdfp6k2wml9s3ssyfwnajncqm79-libarchive-3.0.2.1-doc/share/doc/libarchive-3.0.2.1
Configuration files installed in:
/nix/store/jw8fr6ggh8pqlayn7wnwm9hjr1kbkksh-libarchive-3.0.2.1/etc
No alex found
Using ar found on system at:
/nix/store/7m6p435s1c9ysfs266n9ppfw9zkii697-binutils-2.35.1/bin/ar
Using c2hs version 0.28.7 found on system at:
/nix/store/d0q4i8h1652a6zhkhns3scy7p7900npc-c2hs-0.28.7/bin/c2hs
Using cpphs version 1.20.9 found on system at:
/nix/store/1yixns8ac9w2xd17baqglx5i96130m77-cpphs-1.20.9.1/bin/cpphs
No doctest found
Using gcc version 10.3.0 given by user at:
/nix/store/jabh25wi8r7v2gpkfyhvm9jaxpzgw73x-gcc-wrapper-10.3.0/bin/gcc
Using ghc version 8.10.4 found on system at:
/nix/store/jyny464p3c2g9dkq5y2fw2g4jgfvkm33-ghc-8.10.4/bin/ghc
Using ghc-pkg version 8.10.4 found on system at:
/nix/store/jyny464p3c2g9dkq5y2fw2g4jgfvkm33-ghc-8.10.4/bin/ghc-pkg
No ghcjs found
No ghcjs-pkg found
No greencard found
Using haddock version 2.24.0 found on system at:
/nix/store/jyny464p3c2g9dkq5y2fw2g4jgfvkm33-ghc-8.10.4/bin/haddock
No happy found
Using haskell-suite found on system at: haskell-suite-dummy-location
Using haskell-suite-pkg found on system at: haskell-suite-pkg-dummy-location
No hmake found
Using hpc version 0.68 found on system at:
/nix/store/jyny464p3c2g9dkq5y2fw2g4jgfvkm33-ghc-8.10.4/bin/hpc
Using hsc2hs version 0.68.7 found on system at:
/nix/store/jyny464p3c2g9dkq5y2fw2g4jgfvkm33-ghc-8.10.4/bin/hsc2hs
Using hscolour version 1.24 found on system at:
/nix/store/h7kqmh5gpyawqxnszlv054viybyxcpsz-hscolour-1.24.4/bin/HsColour
No jhc found
Using ld found on system at:
/nix/store/2mffrrdds9i27mdms0w7kx5m711y1hni-binutils-wrapper-2.35.1/bin/ld.gold
Using pkg-config version 0.29.2 found on system at:
/nix/store/5nwv4kv0sqhzi7alkvnl8jx3k9mrnhxi-pkg-config-wrapper-0.29.2/bin/pkg-config
Using runghc version 8.10.4 found on system at:
/nix/store/jyny464p3c2g9dkq5y2fw2g4jgfvkm33-ghc-8.10.4/bin/runghc
Using strip version 2.35 found on system at:
/nix/store/7m6p435s1c9ysfs266n9ppfw9zkii697-binutils-2.35.1/bin/strip
Using tar found on system at:
/nix/store/l1qk6wapdkzj4q1dfiqv5vw479bsy0sw-gnutar-1.34/bin/tar
No uhc found
building
Preprocessing library for libarchive-3.0.2.1..
Building library for libarchive-3.0.2.1..
[ 1 of 15] Compiling Codec.Archive.Types.Foreign ( dist/build/Codec/Archive/Types/Foreign.hs, dist/build/Codec/Archive/Types/Foreign.o, dist/build/Codec/Archive/Types/Foreign.dyn_o )

src/Codec/Archive/Types/Foreign.chs:45:1: warning: [-Wunused-imports]
    The import of ‘Data.Semigroup’ is redundant
      except perhaps to import instances from ‘Data.Semigroup’
    To import instances alone, use: import Data.Semigroup()
   |
45 | import           Foreign.Ptr        (Ptr)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ 2 of 15] Compiling Codec.Archive.Types ( src/Codec/Archive/Types.hs, dist/build/Codec/Archive/Types.o, dist/build/Codec/Archive/Types.dyn_o )
[ 3 of 15] Compiling Codec.Archive.Permissions ( src/Codec/Archive/Permissions.hs, dist/build/Codec/Archive/Permissions.o, dist/build/Codec/Archive/Permissions.dyn_o )
[ 4 of 15] Compiling Codec.Archive.Internal.Pack.Common ( src/Codec/Archive/Internal/Pack/Common.hs, dist/build/Codec/Archive/Internal/Pack/Common.o, dist/build/Codec/Archive/Internal/Pack/Common.dyn_o )
[ 5 of 15] Compiling Codec.Archive.Internal.Monad ( src/Codec/Archive/Internal/Monad.hs, dist/build/Codec/Archive/Internal/Monad.o, dist/build/Codec/Archive/Internal/Monad.dyn_o )
[ 6 of 15] Compiling Codec.Archive.Foreign.ArchiveEntry.Macros ( dist/build/Codec/Archive/Foreign/ArchiveEntry/Macros.hs, dist/build/Codec/Archive/Foreign/ArchiveEntry/Macros.o, dist/build/Codec/Archive/Foreign/ArchiveEntry/Macros.dyn_o )
[ 7 of 15] Compiling Codec.Archive.Foreign.Archive.Macros ( dist/build/Codec/Archive/Foreign/Archive/Macros.hs, dist/build/Codec/Archive/Foreign/Archive/Macros.o, dist/build/Codec/Archive/Foreign/Archive/Macros.dyn_o )
[ 8 of 15] Compiling Codec.Archive.Foreign.Archive ( dist/build/Codec/Archive/Foreign/Archive.hs, dist/build/Codec/Archive/Foreign/Archive.o, dist/build/Codec/Archive/Foreign/Archive.dyn_o )

src/Codec/Archive/Foreign/Archive.chs:342:1: warning: [-Wunused-imports]
    The qualified import of ‘Codec.Archive.Types.Foreign’ is redundant
      except perhaps to import instances from ‘Codec.Archive.Types.Foreign’
    To import instances alone, use: import Codec.Archive.Types.Foreign()
    |
342 | import Foreign.Marshal.Alloc (alloca)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ 9 of 15] Compiling Codec.Archive.Foreign.ArchiveEntry ( dist/build/Codec/Archive/Foreign/ArchiveEntry.hs, dist/build/Codec/Archive/Foreign/ArchiveEntry.o, dist/build/Codec/Archive/Foreign/ArchiveEntry.dyn_o )
[10 of 15] Compiling Codec.Archive.Foreign ( src/Codec/Archive/Foreign.hs, dist/build/Codec/Archive/Foreign.o, dist/build/Codec/Archive/Foreign.dyn_o )
[11 of 15] Compiling Codec.Archive.Internal.Unpack ( src/Codec/Archive/Internal/Unpack.hs, dist/build/Codec/Archive/Internal/Unpack.o, dist/build/Codec/Archive/Internal/Unpack.dyn_o )
[12 of 15] Compiling Codec.Archive.Internal.Unpack.Lazy ( src/Codec/Archive/Internal/Unpack/Lazy.hs, dist/build/Codec/Archive/Internal/Unpack/Lazy.o, dist/build/Codec/Archive/Internal/Unpack/Lazy.dyn_o )
[13 of 15] Compiling Codec.Archive.Internal.Pack ( src/Codec/Archive/Internal/Pack.hs, dist/build/Codec/Archive/Internal/Pack.o, dist/build/Codec/Archive/Internal/Pack.dyn_o )

src/Codec/Archive/Internal/Pack.hs:29:1: warning: [-Wunused-imports]
    The import of ‘forM_’ from module ‘Control.Monad’ is redundant
   |
29 | import           Control.Monad                      (forM_, void)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Codec/Archive/Internal/Pack.hs:33:1: warning: [-Wunused-imports]
    The qualified import of ‘Data.ByteString.Lazy’ is redundant
      except perhaps to import instances from ‘Data.ByteString.Lazy’
    To import instances alone, use: import Data.ByteString.Lazy()
   |
33 | import qualified Data.ByteString.Lazy               as BSL
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Codec/Archive/Internal/Pack.hs:38:1: warning: [-Wunused-imports]
    The import of ‘CLLong’ from module ‘Foreign.C.Types’ is redundant
   |
38 | import           Foreign.C.Types                    (CLLong (..), CLong (..))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[14 of 15] Compiling Codec.Archive.Internal.Pack.Lazy ( src/Codec/Archive/Internal/Pack/Lazy.hs, dist/build/Codec/Archive/Internal/Pack/Lazy.o, dist/build/Codec/Archive/Internal/Pack/Lazy.dyn_o )
[15 of 15] Compiling Codec.Archive    ( src/Codec/Archive.hs, dist/build/Codec/Archive.o, dist/build/Codec/Archive.dyn_o )
[ 1 of 15] Compiling Codec.Archive.Types.Foreign ( dist/build/Codec/Archive/Types/Foreign.hs, dist/build/Codec/Archive/Types/Foreign.p_o )

src/Codec/Archive/Types/Foreign.chs:45:1: warning: [-Wunused-imports]
    The import of ‘Data.Semigroup’ is redundant
      except perhaps to import instances from ‘Data.Semigroup’
    To import instances alone, use: import Data.Semigroup()
   |
45 | import           Foreign.Ptr        (Ptr)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ 2 of 15] Compiling Codec.Archive.Types ( src/Codec/Archive/Types.hs, dist/build/Codec/Archive/Types.p_o )
[ 3 of 15] Compiling Codec.Archive.Permissions ( src/Codec/Archive/Permissions.hs, dist/build/Codec/Archive/Permissions.p_o )
[ 4 of 15] Compiling Codec.Archive.Internal.Pack.Common ( src/Codec/Archive/Internal/Pack/Common.hs, dist/build/Codec/Archive/Internal/Pack/Common.p_o )
[ 5 of 15] Compiling Codec.Archive.Internal.Monad ( src/Codec/Archive/Internal/Monad.hs, dist/build/Codec/Archive/Internal/Monad.p_o )
[ 6 of 15] Compiling Codec.Archive.Foreign.ArchiveEntry.Macros ( dist/build/Codec/Archive/Foreign/ArchiveEntry/Macros.hs, dist/build/Codec/Archive/Foreign/ArchiveEntry/Macros.p_o )
[ 7 of 15] Compiling Codec.Archive.Foreign.Archive.Macros ( dist/build/Codec/Archive/Foreign/Archive/Macros.hs, dist/build/Codec/Archive/Foreign/Archive/Macros.p_o )
[ 8 of 15] Compiling Codec.Archive.Foreign.Archive ( dist/build/Codec/Archive/Foreign/Archive.hs, dist/build/Codec/Archive/Foreign/Archive.p_o )

src/Codec/Archive/Foreign/Archive.chs:342:1: warning: [-Wunused-imports]
    The qualified import of ‘Codec.Archive.Types.Foreign’ is redundant
      except perhaps to import instances from ‘Codec.Archive.Types.Foreign’
    To import instances alone, use: import Codec.Archive.Types.Foreign()
    |
342 | import Foreign.Marshal.Alloc (alloca)
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[ 9 of 15] Compiling Codec.Archive.Foreign.ArchiveEntry ( dist/build/Codec/Archive/Foreign/ArchiveEntry.hs, dist/build/Codec/Archive/Foreign/ArchiveEntry.p_o )
[10 of 15] Compiling Codec.Archive.Foreign ( src/Codec/Archive/Foreign.hs, dist/build/Codec/Archive/Foreign.p_o )
[11 of 15] Compiling Codec.Archive.Internal.Unpack ( src/Codec/Archive/Internal/Unpack.hs, dist/build/Codec/Archive/Internal/Unpack.p_o )
[12 of 15] Compiling Codec.Archive.Internal.Unpack.Lazy ( src/Codec/Archive/Internal/Unpack/Lazy.hs, dist/build/Codec/Archive/Internal/Unpack/Lazy.p_o )
[13 of 15] Compiling Codec.Archive.Internal.Pack ( src/Codec/Archive/Internal/Pack.hs, dist/build/Codec/Archive/Internal/Pack.p_o )

src/Codec/Archive/Internal/Pack.hs:29:1: warning: [-Wunused-imports]
    The import of ‘forM_’ from module ‘Control.Monad’ is redundant
   |
29 | import           Control.Monad                      (forM_, void)
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Codec/Archive/Internal/Pack.hs:33:1: warning: [-Wunused-imports]
    The qualified import of ‘Data.ByteString.Lazy’ is redundant
      except perhaps to import instances from ‘Data.ByteString.Lazy’
    To import instances alone, use: import Data.ByteString.Lazy()
   |
33 | import qualified Data.ByteString.Lazy               as BSL
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

src/Codec/Archive/Internal/Pack.hs:38:1: warning: [-Wunused-imports]
    The import of ‘CLLong’ from module ‘Foreign.C.Types’ is redundant
   |
38 | import           Foreign.C.Types                    (CLLong (..), CLong (..))
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[14 of 15] Compiling Codec.Archive.Internal.Pack.Lazy ( src/Codec/Archive/Internal/Pack/Lazy.hs, dist/build/Codec/Archive/Internal/Pack/Lazy.p_o )
[15 of 15] Compiling Codec.Archive    ( src/Codec/Archive.hs, dist/build/Codec/Archive.p_o )
Preprocessing test suite 'libarchive-test' for libarchive-3.0.2.1..
Building test suite 'libarchive-test' for libarchive-3.0.2.1..
[1 of 3] Compiling Codec.Archive.Roundtrip ( test/Codec/Archive/Roundtrip.hs, dist/build/libarchive-test/libarchive-test-tmp/Codec/Archive/Roundtrip.o )
[2 of 3] Compiling Codec.Archive.Test ( test/Codec/Archive/Test.hs, dist/build/libarchive-test/libarchive-test-tmp/Codec/Archive/Test.o )
[3 of 3] Compiling Main             ( dist/build/libarchive-test/libarchive-test-tmp/Spec.hs, dist/build/libarchive-test/libarchive-test-tmp/Main.o )
Linking dist/build/libarchive-test/libarchive-test ...
running tests
Running 1 test suites...
Test suite libarchive-test: RUNNING...

roundtrip
  sucessfully unpacks/packs (test/data/aarch64-linux-dist.tar)
  should repack (zip/test/data/aarch64-linux-dist.tar)
  should repack (cpio)
  should repack (shar)
  should repack (7zip)
  works on nonstandard bytestring (test/data/aarch64-linux-dist.tar)
  works on strict bytestring (test/data/aarch64-linux-dist.tar)
  works on strict bytestring (test/data/aarch64-linux-dist.tar)
  unpacks test/data/aarch64-linux-dist.tar
  reads test/data/aarch64-linux-dist.tar
  with symlinks
    packs/unpacks successfully without loss
    packs/unpacks on filesystem successfully without loss FAILED [1]
  with hardlinks
    packs/unpacks successfully without loss
    issue#4
      packs/unpacks on filesystem successfully without loss
  with forward referenced hardlinks
    packs/unpacks successfully without loss
    re-ordering on unpack
      packs/unpacks on filesystem successfully without loss
        # PENDING: No reason given
  having entry without ownership
    packs/unpacks successfully without loss
      # PENDING: No reason given
  having entry without timestamp
    packs/unpacks successfully without loss
      # PENDING: No reason given

Failures:

  test/Codec/Archive/Roundtrip.hs:65:59: 
  1) roundtrip, with symlinks, packs/unpacks on filesystem successfully without loss
       uncaught exception: IOException of type NoSuchThing
       ./x/b.txt: getFileStatus: does not exist (No such file or directory)

  To rerun use: --match "/roundtrip/with symlinks/packs/unpacks on filesystem successfully without loss/"

Randomized with seed 120881912

Finished in 7.5625 seconds
18 examples, 1 failure, 3 pending
Test suite libarchive-test: FAIL
Test suite logged to: dist/test/libarchive-3.0.2.1-libarchive-test.log
0 of 1 test suites (0 of 1 test cases) passed.
```

**Comment by sternenseemann** [comment]
Been told that this happens for other as well for tests executed with >= 12 jobs, but not as reliable as in our case.

**Comment by vmchale** [comment]
Unfortunately I've gotten that as well (sporadically), not sure why but I'll have a look.

Flaky test suite on my part I think.

**Comment by vmchale** [comment]
Haven't been able to reproduce with GHC 9.0.1 (x86)

**Comment by sternenseemann** [comment]
Only tested GHC 8.10.4, I guess I’ll try again when we have switched to that as the default version.

**Comment by hasufell** [comment]
I believe this should be fixed in https://github.com/vmchale/libarchive/commit/82d7f6a99595440602987088ec24c96bb0f60e89#diff-0a195188bdcf4bdc24fe5ec180dbcf24e4ca482168ffc057ba71c55eceae0b99
